packages/console/src/ds-components/TextInput/MultiInput.tsx

```diff
@@ -0,0 +1,171 @@
+import { type AdminConsoleKey } from '@logto/phrases';
+import { generateStandardShortId } from '@logto/shared/universal';
+import { type Nullable, conditional } from '@silverhand/essentials';
+import classNames from 'classnames';
+import { useEffect, useMemo, useRef, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+
+import Close from '@/assets/icons/close.svg';
+import { onKeyDownHandler } from '@/utils/a11y';
+
+import IconButton from '../IconButton';
+import Tag, { type Props as TagProps } from '../Tag';
+
+import * as styles from './MultiInput.module.scss';
+
+export type Option = {
+  /**
+   * Generate a random unique id for each option to handle deletion.
+   * Sometimes we may have options with the same value, which is allowed when inputting but prohibited when submitting.
+   */
+  id: string;
+  value: string;
+  /**
+   * The `status` is used to indicate the status of the domain item (could fall into following categories):
+   * - undefined: valid domain
+   * - 'info': duplicated domain or blocked domain {@link @logto/core/src/sso/index.ts}
+   */
+  status?: Extract<TagProps['status'], 'info'>;
+};
+
+type Props = {
+  className?: string;
+  values: Option[];
+  onChange: (values: Option[]) => void;
+  error?: string | boolean;
+  placeholder?: AdminConsoleKey;
+  renderOption?: (option: Option) => React.ReactNode;
+};
+
+// RegExp to domain string.
+// eslint-disable-next-line prefer-regex-literals
+const domainRegExp = new RegExp('\\S+[\\.]{1}\\S+');
+
+function MultiInput({
```

Comment:
@xiaoyijun can you take a look at this component?

Reply:
This component seems not a design system component. Need to align with designers.

Reply:
Try to implement base on the `MultiSelect` component? since the are similar.

Reply:
Good call, move to `EnterpriseSsoDetails`